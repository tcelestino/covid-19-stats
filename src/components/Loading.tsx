import React from 'react';
import { CircularProgress, createStyles, makeStyles } from '@material-ui/core';

const useStyles = makeStyles(() =>
  createStyles({
    root: {
      display: 'flex',
      justifyContent: 'center',
      alignItems: 'center',
      minHeight: '40vh',
    },
  })
);

function Loading(): JSX.Element {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <CircularProgress color='secondary' />
    </div>
  );
}

export default Loading;
