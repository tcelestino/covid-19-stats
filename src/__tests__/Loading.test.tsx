import React from 'react';
import { render } from '@testing-library/react';
import Loading from '../components/Loading';

describe('<Loading />', () => {
  it('render page', () => {
    const { container } = render(<Loading />);
    expect(container).toMatchSnapshot();
  });
});
