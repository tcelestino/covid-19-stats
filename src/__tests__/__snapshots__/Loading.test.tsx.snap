// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> render page 1`] = `
<div>
  <div
    class="makeStyles-root-1"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorSecondary MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <h1
          class="MuiTypography-root makeStyles-title-2 MuiTypography-h1"
        >
          <a
            class="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-titleText-3 MuiTypography-colorPrimary"
            href="//tcelestino.github.io/covid-19-stats/"
          >
            COVID-19 Stats
          </a>
        </h1>
      </div>
    </header>
    <main
      class="MuiContainer-root MuiContainer-maxWidthSm"
    >
      <div
        class="MuiFormControl-root makeStyles-form-92 MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
          data-shrink="false"
          id="demo-simple-select-autowidth-label"
        >
          Choose a country
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-select-93 MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
        >
          <div
            aria-haspopup="listbox"
            aria-labelledby=" "
            class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
            role="button"
            tabindex="0"
          >
            <span>
              â€‹
            </span>
          </div>
          <input
            type="hidden"
            value=""
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSelect-icon"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
      <div
        class="makeStyles-root-174"
      >
        <div
          class="MuiCircularProgress-root MuiCircularProgress-colorSecondary MuiCircularProgress-indeterminate"
          role="progressbar"
          style="width: 40px; height: 40px;"
        >
          <svg
            class="MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
              cx="44"
              cy="44"
              fill="none"
              r="20.2"
              stroke-width="3.6"
            />
          </svg>
        </div>
      </div>
      
    </main>
    <footer
      class="makeStyles-footer-4"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthSm"
      >
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          Information data is supplied by
           
          <a
            href="https://thevirustracker.com/api"
            rel="noopener noreferrer"
            target="_blank"
          >
            Coronavirus Tracker
          </a>
        </p>
      </div>
    </footer>
  </div>
</div>
`;

exports[`<Loading /> render page 1`] = `
<div>
  <div
    class="makeStyles-root-1"
  >
    <div
      class="MuiCircularProgress-root MuiCircularProgress-colorSecondary MuiCircularProgress-indeterminate"
      role="progressbar"
      style="width: 40px; height: 40px;"
    >
      <svg
        class="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </div>
  </div>
</div>
`;
